File: frontend/lib/weather.ts
import type { WeatherLabel } from "./api";

/** Emoji / symbol for weather — keep simple for cross-font support */
export const WEATHER_GLYPH: Record<WeatherLabel, string> = {
  Momentum: "☀",
  Confusion: "☁",
  Burnout: "⛈",
  Overload: "🌧",
  Recovery: "🌤"
};

export const WEATHER_LABEL_ZH: Record<WeatherLabel, string> = {
  Momentum: "顺势",
  Confusion: "迷雾",
  Burnout: "耗竭",
  Overload: "过载",
  Recovery: "回升"
};

export const WEATHER_BLURB: Record<WeatherLabel, string> = {
  Momentum: "你在节奏里，尽量护住它。",
  Confusion: "有些不清楚也没关系，允许自己停在这里。",
  Burnout: "累了就先当正事休息，不必愧疚。",
  Overload: "输入太多时，先关掉一条支流。",
  Recovery: "能回来一点点就很难得，保持小事就好。"
};

File: cli/weatherflow_cli/weather.py
"""``wf weather`` — print current life weather."""

from __future__ import annotations

import typer
from rich.console import Console
from rich.panel import Panel

from weatherflow_cli import api

console = Console()


_LABEL_EMOJI = {
    "Momentum": "?",
    "Confusion": "?",
    "Burnout": "??",
    "Overload": "?",
    "Recovery": "??",
}


def run() -> None:
    try:
        state = api.get("/api/state/current")
    except Exception:
        console.print(
            "[yellow]No state yet. Try [bold]wf checkin[/bold] first.[/yellow]"
        )
        raise typer.Exit(code=1)

    label = state.get("weather_label", "Confusion")
    emoji = _LABEL_EMOJI.get(label, "·")
    console.print(
        Panel.fit(
            f"[bold]{emoji}  {label}[/bold]   "
            f"focus={state['focus']} momentum={state['momentum']} "
            f"burnout={state['burnout']} stress={state['stress']} "
            f"confidence={state['confidence']} motivation={state['motivation']}\n"
            f"[dim]{state.get('rationale') or ''}[/dim]",
            title="Life weather",
            border_style="cyan",
        )
    )
